src/CliMenu.php

```diff
@@ -75,40 +84,33 @@ public function __construct(
 
     /**
      * Configure the terminal to work with CliMenu
-     *
-     * @throws InvalidTerminalException
      */
     protected function configureTerminal() : void
     {
         $this->assertTerminalIsValidTTY();
 
-        $this->terminal->setCanonicalMode();
+        $this->terminal->disableCanonicalMode();
+        $this->terminal->disableEchoBack();
         $this->terminal->disableCursor();
         $this->terminal->clear();
     }
 
     /**
      * Revert changes made to the terminal
-     *
-     * @throws InvalidTerminalException
      */
     protected function tearDownTerminal() : void
     {
-        $this->assertTerminalIsValidTTY();
-
-        $this->terminal->setCanonicalMode(false);
-        $this->terminal->enableCursor();
+        $this->terminal->restoreOriginalConfiguration();
     }
 
     private function assertTerminalIsValidTTY() : void
     {
-        if (!$this->terminal->isTTY()) {
-            throw new InvalidTerminalException(
-                sprintf('Terminal "%s" is not a valid TTY', $this->terminal->getDetails())
-            );
+        if (!$this->terminal->isInteractive()) {
+            throw new InvalidTerminalException('Terminal is not interactive (TTY)');
```

Comment:
This is what I mentioned in the `php-school/terminal` PR. If it's not a valid TTY - then there is no name.